packages/react-aria/src/hooks/useAriaButton.ts

```diff
@@ -0,0 +1,94 @@
+import * as React from 'react';
+import { ObjectShorthandProps, useEventCallback } from '@fluentui/react-utilities';
+
+export function useARIAButton(shorthand: ObjectShorthandProps<React.ButtonHTMLAttributes<HTMLElement>>) {
+  if (!shorthand.hasOwnProperty('as')) {
+    shorthand.as = 'button';
+  }
+  if (!shorthand.hasOwnProperty('children')) {
+    shorthand.children = null;
+  }
+
+  const { onClick, onKeyDown, onKeyUp, as, disabled } = shorthand;
+
+  const onClickHandler = useEventCallback((ev: React.MouseEvent<HTMLElement>) => {
+    if (disabled) {
+      ev.preventDefault();
+    } else {
+      onClick?.(ev);
+    }
+  });
+
+  const onKeyDownHandler = useEventCallback((ev: React.KeyboardEvent<HTMLElement>) => {
+    onKeyDown?.(ev);
+    if (ev.isDefaultPrevented()) {
+      return;
+    }
+    if (disabled && (ev.key === 'Enter' || ev.key === ' ')) {
+      ev.preventDefault();
+      ev.stopPropagation();
+      return;
+    }
+    if (ev.key === ' ') {
+      ev.preventDefault();
+    }
+    // If enter is pressed, activate the button
+    else if (ev.key === 'Enter') {
+      ev.preventDefault();
+      ev.currentTarget.click();
+    }
+  });
+
+  const onKeyupHandler = useEventCallback((ev: React.KeyboardEvent<HTMLElement>) => {
+    onKeyUp?.(ev);
+    if (ev.isDefaultPrevented()) {
+      return;
+    }
+    if (disabled && (ev.key === 'Enter' || ev.key === ' ')) {
+      ev.preventDefault();
+      ev.stopPropagation();
+      return;
+    }
+    if (ev.key === ' ') {
+      ev.preventDefault();
+      ev.currentTarget.click();
+    }
+  });
+
+  /**
+   * TODO: Ideally this is unnecessary after implementation of as-prop RFC.
+   * The way to go is to have an assertion method to ensure types,
+   * in the case of button we'd like to limit it for: button, div, span, a
+   */
+  if (typeof as === 'string') {
```

Comment:
to avoid unnecessary nesting, especially for a case that shouldn't be supported now, does it make sense to return `undefined as never` early ?

Reply:
🤔... sounds like a plan actually! Haven't thought about that.

Reply:
+1 to @ling1726 - yeah in general exit early strategy is most readable from my experience